### PR TITLE
Basic GCW regions

### PR DESCRIPTION
**Pull request by madsboddum**
- Creation of regions through the sdb.
- Display regions in-game.
- Data on regions is persisted.
- Every 15 minutes presence is detected and factions are rewarded or punished for not being present.
- GCW points earned inside zones count towards percentage. Points earned outside or in space do not (yet) count.
- The GCW window isn't working 100%, but at least it doesn't crash the client anymore.

### CODE REVIEW

File: src/main/java/com/projectswg/holocore/resources/support/data/server_info/mongodb/PswgGcwRegionDatabase.kt [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+	fun createZone(zoneName: String) {
+		collection ?: return
+
+		val mongoData = MongoData()
+
+		// Initiualize a zone with some points. We do this so gaining 100% control can't be done by being the first to receive any amount of GCW points.
+		mongoData.putString("zone", zoneName)
+		mongoData.putLong("imperialPoints", CivilWarRegionService.BASE_ZONE_POINTS)
+		mongoData.putLong("rebelPoints", CivilWarRegionService.BASE_ZONE_POINTS)
```
Comment (Josh-Larson) [inline src/main/java/com/projectswg/holocore/resources/support/data/server_info/mongodb/PswgGcwRegionDatabase.kt]: I don't like this database to service coupling, can you pass them in as parameters?

File: src/main/java/com/projectswg/holocore/services/gameplay/gcw/CivilWarRegionService.java [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+			}
+			
+			Set<Player> observers = egg.getObservers();
+			
+			long rebels = countPresentPlayers(observers, PvpFaction.REBEL);
+			long rebelPoints = zoneMetadata.getRebelPoints();
+			
+			if (rebels <= 0) {
+				// No rebels present. Deduct points.
+				rebelPoints =  Math.min(zoneMetadata.getRebelPoints() - PRESENCE_POINTS, BASE_ZONE_POINTS);
```
Comment (Josh-Larson) [inline src/main/java/com/projectswg/holocore/services/gameplay/gcw/CivilWarRegionService.java]: Shouldn't this be max?

Comment (madsboddum) [reply src/main/java/com/projectswg/holocore/services/gameplay/gcw/CivilWarRegionService.java]: Yep, you're right.

File: src/main/java/com/projectswg/holocore/resources/support/data/server_info/mongodb/PswgGcwRegionDatabase.kt (hunk 1 above)
Comment (madsboddum) [reply src/main/java/com/projectswg/holocore/resources/support/data/server_info/mongodb/PswgGcwRegionDatabase.kt]: Good point.

**Pull request merged**